Warnings when using with Frosting

When using this addin with Frosting a warning is shown.

Warning in Visual Studio:

```
Warning	MSB3246	Resolved file has a bad image, no metadata, or is otherwise inaccessible. Could not load file or assembly 'git2-6777db8.dll' or one of its dependencies. The module was expected to contain an assembly manifest.	Cake.Frosting.Issues.Recipe	C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets	2123	
```

Warning when building with dotnet:

```
C:\Program Files\dotnet\sdk\5.0.102\Microsoft.Common.CurrentVersion.targets(2123,5): warning MSB3246: Resolved file has a bad image, no metadata, or is otherwise inaccessible. Assembly file 'C:\Users\PascalBerger\.nuget\packages\cake.git\1.0.0\lib\netstandard2.0\git2-6777db8.dll' could not be opened -- PE image doesn't contain managed metadata. [C:\git\github\Cake.Issues.Recipe\Cake.Frosting.Issues.Recipe\Cake.Frosting.Issues.Recipe\Cake.Frosting.Issues.Recipe.csproj]
```

--- a/build.cake
+++ b/build.cake
@@ -90,11 +90,7 @@ Setup(ctx =>
                                 Title                   = assemblyInfo.Title,
                                 Authors                 = new[] {assemblyInfo.Company},
                                 Owners                  = new[] {assemblyInfo.Company, "cake-contrib"},
-                                Description             = assemblyInfo.Description + 
-                                                          Environment.NewLine + Environment.NewLine + 
-                                                          "NOTE:" + 
-                                                          Environment.NewLine + 
-                                                          "The addin currently only runs on x64 processors. ARM processors are not supported.",
+                                Description             = assemblyInfo.Description,
                                 Summary                 = "Cake AddIn that extends Cake with Git SCM features",
                                 ProjectUrl              = new Uri("https://github.com/cake-contrib/Cake_Git/"),
                                 Repository              = new NuGetRepository { Type = "git", Url = "https://github.com/cake-contrib/Cake_Git.git" },
@@ -116,6 +112,7 @@ Setup(ctx =>
             artifactsRoot,
             nugetRoot,
             nugetRoot.CombineWithFilePath("Cake.Git." + semVersion + ".nupkg"),
+            nugetRoot.CombineWithFilePath("Cake.Frosting.Git." + semVersion + ".nupkg"),
             GetFiles("./**/*.sln")
                 .ToArray(),
             MakeAbsolute(Directory("./tools/Addins/Cake.Git/Cake.Git"))));
@@ -214,6 +211,10 @@ Task("Publish-Artifacts")
 
 
 Task("Create-NuGet-Package")
+    .IsDependentOn("Create-NuGet-Package-Scripting")
+    .IsDependentOn("Create-NuGet-Package-Frosting");
+    
+Task("Create-NuGet-Package-Scripting")
     .IsDependentOn("Publish-Artifacts")
     .Does<BuildData>(static (context, data) =>
 {
@@ -222,6 +223,11 @@ Task("Create-NuGet-Package")
     var libGit = context.GetFiles(data.BuildPaths.ArtifactsRoot.FullPath + "/**/LibGit2Sharp*");
     var unmanaged = context.GetFiles(data.BuildPaths.ArtifactsRoot.FullPath + "/net6.0/runtimes/**/*");
 
+    data.NuGetPackSettings.Description += Environment.NewLine + Environment.NewLine + 
+                                          "NOTE:" + Environment.NewLine + 
+                                          "The addin currently only runs on x64 processors. ARM processors are not supported." + Environment.NewLine +
+                                          "This is the version of the addin compatible with Cake Script Runners." + Environment.NewLine +
+                                          "For addin compatible with Cake Frosting see Cake.Frosting.Git.";
     data.NuGetPackSettings.Files =  (libGit + cakeGit + cakeGitDoc)
                                     .Select(file=>file.FullPath.Substring(data.BuildPaths.ArtifactsRoot.FullPath.Length+1))
                                     .Select(file=>new NuSpecContent {Source = file, Target = "lib/" + file})
@@ -253,8 +259,44 @@ Task("Create-NuGet-Package")
     context.NuGetPack(data.NuGetPackSettings);
 });
 
+Task("Create-NuGet-Package-Frosting")
+    .IsDependentOn("Publish-Artifacts")
+    .Does<BuildData>(static (context, data) =>
+{
+    var cakeGit = context.GetFiles(data.BuildPaths.ArtifactsRoot.FullPath + "/**/Cake.Git.dll");
+    var cakeGitDoc = context.GetFiles(data.BuildPaths.ArtifactsRoot.FullPath + "/**/Cake.Git.xml");
+
+    data.NuGetPackSettings.Id = "Cake.Frosting.Git";
+    data.NuGetPackSettings.Description += Environment.NewLine + Environment.NewLine + 
+                                          "NOTE:" + Environment.NewLine + 
+                                          "This is the version of the addin compatible with Cake Frosting." + Environment.NewLine +
+                                          "For addin compatible with Cake Script Runners see Cake.Git." + Environment.NewLine;
+    data.NuGetPackSettings.Files =  (cakeGit + cakeGitDoc)
+                                    .Select(file=>file.FullPath.Substring(data.BuildPaths.ArtifactsRoot.FullPath.Length+1))
+                                    .Select(file=>new NuSpecContent {Source = file, Target = "lib/" + file})
+                                    // add the icon
+                                    .Union(new []
+                                    {
+                                        new NuSpecContent
+                                        {
+                                            Source = context.MakeAbsolute(context.File("./asset/cake-contrib-addin-medium.png")).FullPath,
+                                            Target = "images/icon.png",
+                                        },
+                                    })
+                                    .ToArray();
+    data.NuGetPackSettings.Dependencies = new List<NuSpecDependency>
+    {
+        new NuSpecDependency { Id = "Cake.Core", Version = "4.0.0" },
+        new NuSpecDependency { Id = "LibGit2Sharp", Version = "0.29.0" }
+    };
+
+    context.EnsureDirectoryExists(data.BuildPaths.NuGetRoot);
+
+    context.NuGetPack(data.NuGetPackSettings);
+});
+
 Task("Test")
-    .IsDependentOn("Create-NuGet-Package")
+    .IsDependentOn("Create-NuGet-Package-Scripting")
     .WithCriteria<BuildData>(data => StringComparer.OrdinalIgnoreCase.Equals(data.Configuration, "Release"))
     .DoesForEach<BuildData, TestCase>(
         static (data, context) => {
@@ -266,7 +308,7 @@ Task("Test")
                 });
             }
 
-            context.Unzip(data.BuildPaths.Package, data.BuildPaths.AddinDir);
+            context.Unzip(data.BuildPaths.ScriptingPackage, data.BuildPaths.AddinDir);
 
             var cakeSettings = new CakeSettings {
                 Arguments = new Dictionary<string, string> {
@@ -296,7 +338,8 @@ Task("Upload-AppVeyor-Artifacts")
     .Does<BuildData>(data =>
 {
     // Upload Artifact
-    AppVeyor.UploadArtifact(data.BuildPaths.Package);
+    AppVeyor.UploadArtifact(data.BuildPaths.ScriptingPackage);
+    AppVeyor.UploadArtifact(data.BuildPaths.FrostingPackage);
 });
 
 Task("Push-NuGet-Packages")
@@ -306,7 +349,15 @@ Task("Push-NuGet-Packages")
     .Does<BuildData>((context, data) =>
 {
    context.DotNetNuGetPush(
-                data.BuildPaths.Package,
+                data.BuildPaths.ScriptingPackage,
+                new DotNetNuGetPushSettings
+                {
+                    Source = data.NuGetSource,
+                    ApiKey = data.NuGetApiKey
+                }
+        );
+   context.DotNetNuGetPush(
+                data.BuildPaths.FrostingPackage,
                 new DotNetNuGetPushSettings
                 {
                     Source = data.NuGetSource,
@@ -325,7 +376,8 @@ Task("Upload-GitHubActions-Artifacts")
     var suffix = EnvironmentVariable("matrix-os")
         ?? GitHubActions.Environment.Runner.OS;
     Information("Uploading artifacts for {0}...", suffix);
-    await GitHubActions.Commands.UploadArtifact(data.BuildPaths.Package, string.Concat("NuGet-", suffix));
+    await GitHubActions.Commands.UploadArtifact(data.BuildPaths.ScriptingPackage, string.Concat("Scripting-NuGet-", suffix));
+    await GitHubActions.Commands.UploadArtifact(data.BuildPaths.FrostingPackage, string.Concat("Frosting-NuGet-", suffix));
     await GitHubActions.Commands.UploadArtifact(data.BuildPaths.ArtifactsRoot, string.Concat("Artifacts-", suffix));
 });
 
@@ -359,7 +411,8 @@ RunTarget(target);
 public record BuildPaths(
     DirectoryPath ArtifactsRoot,
     DirectoryPath NuGetRoot,
-    FilePath Package,
+    FilePath ScriptingPackage,
+    FilePath FrostingPackage,
     ICollection<FilePath> Solutions,
     DirectoryPath AddinDir)
 {

